desktop_version/src/Map.cpp

```diff
@@ -412,19 +414,44 @@ void mapclass::roomnamechange(const int x, const int y, const char** lines, cons
     specialroomnames.push_back(roomname);
 }
 
+static bool compareTeleporterPoints(SDL_Point a, SDL_Point b)
+{
+    if (a.x == b.x)
+    {
+        return a.y < b.y;
+    }
+    return a.x < b.x;
+}
+
 void mapclass::initcustommapdata(void)
 {
     shinytrinkets.clear();
+    teleporters.clear();
+
+    std::vector<SDL_Point> teleporters;
```

Comment:
This code works but you should not be shadowing the class member attribute, it impacts readability. Please rename this to `customteleporters`.